src/DataTable/utils/getVisibleColumns.jsx

```diff
@@ -11,13 +12,14 @@ export const selectColumn = {
   // Proptypes disabled as these props are passed in separately
   /* eslint-disable-next-line react/prop-types */
   Header: ({ getToggleAllPageRowsSelectedProps, getToggleAllRowsSelectedProps, page }) => {
+    const { isSelectable, maxSelectableRows } = useContext(DataTableContext);
     const toggleRowsSelectedProps = useMemo(
       () => {
         // determine if this selection is for an individual page or the entire table
         const getToggleRowsSelectedProps = page ? getToggleAllPageRowsSelectedProps : getToggleAllRowsSelectedProps;
-        return getToggleRowsSelectedProps();
+        return isSelectable && maxSelectableRows ? { ...getToggleRowsSelectedProps(), disabled: 'disabled', onChange: () => {} } : getToggleRowsSelectedProps();
```

Comment:
Yep, that is the goal to disabled the `CheckboxControl`  but also if the checkbox is marked not allow to doing anything because someone can modify `disabled`  attribute in the DOM  prop and try to mark the checkbox